llvm/lib/CodeGen/PeepholeOptimizer.cpp

```diff
@@ -220,6 +220,58 @@ class CopyRewriter : public Rewriter {
   }
 };
 
+/// Rewriter for REG_SEQUENCE source operands.  This class is used for
+/// rewriting uses of REG_SEQUENCE operands in arbitrary instructions
+/// whereas the RegSequenceRewriter is used for rewriting REG_SEQUENCE
+/// instructions.
```

Comment:
I'm looking at a similar problem now, and I think there's just a deeper bug in subregister tracking in ValueTracker. I don't think there needs to be a dedicated fold into users of reg_sequences